int64  i_old
int64  j_old
int64  i_new
int64  j_new
---
bool   res
